server/src/main/java/io/pravega/schemaregistry/storage/client/TableStore.java

```diff
@@ -0,0 +1,392 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.storage.client;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.netty.buffer.ByteBuf;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.netty.impl.ConnectionFactoryImpl;
+import io.pravega.client.tables.impl.IteratorState;
+import io.pravega.client.tables.impl.KeyVersion;
+import io.pravega.client.tables.impl.KeyVersionImpl;
+import io.pravega.client.tables.impl.TableEntry;
+import io.pravega.client.tables.impl.TableEntryImpl;
+import io.pravega.client.tables.impl.TableKey;
+import io.pravega.client.tables.impl.TableKeyImpl;
+import io.pravega.common.Exceptions;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.common.tracing.RequestTag;
+import io.pravega.common.util.ContinuationTokenAsyncIterator;
+import io.pravega.common.util.RetriesExhaustedException;
+import io.pravega.common.util.Retry;
+import io.pravega.controller.server.SegmentHelper;
+import io.pravega.controller.server.WireCommandFailedException;
+import io.pravega.controller.store.host.HostStoreException;
+import io.pravega.controller.store.stream.PravegaTablesStoreHelper;
+import io.pravega.controller.store.stream.StoreException;
+import io.pravega.controller.util.RetryHelper;
+import io.pravega.schemaregistry.server.rest.ServiceConfig;
+import io.pravega.schemaregistry.storage.StoreExceptions;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static io.pravega.controller.server.WireCommandFailedException.Reason.ConnectionDropped;
+import static io.pravega.controller.server.WireCommandFailedException.Reason.ConnectionFailed;
+
+/**
+ * Wrapper class over {@link PravegaTablesStoreHelper}. Its implementation abstracts the caller classes from the library
+ * used for interacting with pravega tables.
+ */
+@Slf4j
+public class TableStore {
+    public static final String SCHEMA_REGISTRY_SCOPE = "_schemaregistry";
+    private static final int NUM_OF_RETRIES = 15; // approximately 1 minute worth of retries
+    private final SegmentHelper segmentHelper;
+    private final HostStoreImpl hostStore;
+    private final int numOfRetries;
+    private final ScheduledExecutorService executor;
+    private final Cache<TableCacheKey, Version.VersionedRecord<?>> cache;
```

Comment:
The intent of these instance variables is not automatically clear. Please add some comments to explain them.